modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/task/ScheduleAppointment.java

```diff
@@ -0,0 +1,97 @@
+        value = "The schedule function as any type") })
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "onError",
+        value = "The error function as any type") })
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "appointmentScheduler",
+        value = "The struct with required attributes") })
+@BallerinaAnnotation(annotationName = "Return", attributes = {@Attribute(name = "int)",
+        value = "The identifier of the task") })
+@BallerinaAnnotation(annotationName = "Return", attributes = {@Attribute(name = "any)",
+        value = "The error which is occurred while scheduling the task") })
+public class ScheduleAppointment extends AbstractNativeFunction {
+    private static final Log log = LogFactory.getLog(ScheduleAppointment.class.getName());
+
+    public BValue[] execute(Context ctx) {
+        FunctionRefCPEntry onTriggerFunctionRefCPEntry = null;
+        FunctionRefCPEntry onErrorFunctionRefCPEntry = null;
+        if (ctx.getControlStackNew().getCurrentFrame().getRefLocalVars()[0] != null && ctx.getControlStackNew()
+                .getCurrentFrame().getRefLocalVars()[0] instanceof BFunctionPointer) {
+            onTriggerFunctionRefCPEntry = ((BFunctionPointer) getRefArgument(ctx, 0)).value();
+        }
+        if (ctx.getControlStackNew().getCurrentFrame().getRefLocalVars()[1] != null && ctx.getControlStackNew()
+                .getCurrentFrame().getRefLocalVars()[1] instanceof BFunctionPointer) {
+            onErrorFunctionRefCPEntry = ((BFunctionPointer) getRefArgument(ctx, 1)).value();
+        }
+        BValue scheduler = getRefArgument(ctx, 2);
+        long minute = ((BStruct) scheduler).getIntField(0);
+        long hour = ((BStruct) scheduler).getIntField(1);
+        long dayOfWeek = ((BStruct) scheduler).getIntField(2);
+        long dayOfMonth = ((BStruct) scheduler).getIntField(3);
+        long month = ((BStruct) scheduler).getIntField(4);
+        log.info("Request has come to schedule the appointment with the expression: " + minute + " " + hour + " "
+                + dayOfWeek + " " + dayOfMonth + " " + month);
+        int taskId = -1;
+        BString error = new BString("Unable to schedule the appointment");
+        taskId = TaskUtil.generateTaskId(ctx);
+        if (taskId != -1) {
+            TaskScheduler.triggerAppointment(ctx, taskId, minute, hour, dayOfWeek, dayOfMonth, month,
+                    onTriggerFunctionRefCPEntry, onErrorFunctionRefCPEntry);
+            String schedulerError = StringUtils.isNotEmpty((String) ctx.getProperty(Constant.ERROR + "_" + taskId))
+                    ? ctx.getProperty(Constant.ERROR + "_" + taskId).toString() : "";
+            taskId = StringUtils.isEmpty((String) ctx.getProperty(Constant.ERROR + "_" + taskId)) ? taskId : -1;
+            error = new BString(schedulerError);
+        }
+        return getBValues(new BInteger(taskId), error);
```

Comment:
Do we need to return an error always?

Reply:
Yes, this method returns two fields task id and error

Reply:
But should the error be having a value always? I mean, if there are no errors occurred, shouldn't the value for error be null?